src/db/order-repository.ts

```diff
@@ -0,0 +1,88 @@
+import { Logger } from '../logger';
+import { InitDBResponse } from './db';
+import { from, Observable } from 'rxjs';
+import { Order, Trade } from 'ccxt';
+import { OrderAttributes, OrderInstance } from './order';
+import { map, mergeMap } from 'rxjs/operators';
+import { TradeAttributes } from './trade';
+
+type SaveOrderParams = {
+  order: Order;
+};
+
+class OrderRepository {
+  constructor(private models: InitDBResponse, private logger: Logger) {}
+
+  public saveOrder$ = ({
+    order,
+  }: SaveOrderParams): Observable<OrderInstance> => {
+    this.logger.trace(`Saving order ${JSON.stringify(order)} into database`);
+
+    const arbyOrder = this.convertToArbyOrder(order);
+    const arbyTrades = this.convertToArbyTrades(order.trades, order.id);
+    // TODO async
+    return from(this.models.Order.create(arbyOrder)).pipe(
```

Comment:
Why do we need to use external `this.models` state here? Perhaps we can make due without introducing `OrderRepository` class and pass in the models as an argument?